applications/DG-Max/Algorithms/DivDGMaxEigenvalue.cpp

```diff
@@ -49,24 +49,190 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 using namespace hpgem;
 
+/// Internal storage for the algorithm
+
 template <std::size_t DIM>
-DivDGMaxEigenvalue<DIM>::DivDGMaxEigenvalue(
-    Base::MeshManipulator<DIM>& mesh, std::size_t order,
-    typename DivDGMaxDiscretization<DIM>::Stab stab)
-    : mesh_(mesh), order_(order), stab_(stab) {}
+class DivDGMaxResult;
+
+template <std::size_t DIM>
+struct Workspace {
+
+    friend class DivDGMaxResult<DIM>;
```

Comment:
As noted below, it is really local. Thinking a bit more about it, I think I'll modify it so that it forwards all the methods to the workspace instead of accessing it as a friend.